src/Metadata/AssemblyMetadata.Types.cs

```diff
@@ -6,29 +6,14 @@ namespace Lokad.ILPack.Metadata
 {
     internal partial class AssemblyMetadata
     {
-        public EntityHandle GetTypeHandle(Type type)
+        public EntityHandle GetTypeHandle(Type type, Boolean inMethodBodyWritingContext)
```

Comment:
I would suggest a default value `Boolean inMethodBodyWritingContext = false` which should remove the need to even touch most of the files touched in this commit.